content/docs/bdd/better-gherkin.md

```diff
@@ -80,3 +84,39 @@ Scenario: Subscriber with a paid subscription can access both free and paid arti
   Then she sees a Free article and a Paid article
 ``` 
 With a declarative style, each step communicates an idea, but the exact values aren't specified. The details of *how* the user interacts with the system, such as which specific articles are free or paid, and the subscription level of different test users, are specified in the step definitions (the automation code that interacts with the system). The subscription packages could change in the future. The business could change what content is available to subscribers on free and paid plans, without having to change this scenario and other scenarios that use the same step definitions. If another subscription level is added later, it's easy to add a scenario for that. By avoiding terms like “click a button” that suggest implementation, the scenario is more resilient to implementation details of the UI. The intent of the scenario remains the same, even if the implementation changes later. In addition, having too many implementation details in a scenario, makes it harder to understand the intended behaviour it illustrates.
+
+## A third style 
+
+There is a style that is intermediate between these two.   The exact values are specified in the scenario, but the way the user interacts with the system is not.  This style uses stepdef tables with domain terms as the column headers.   The step definitions can  be reused for mulitple scenarios with different data in the table.  In this example, the scenarios for free and paid subscribers use the same step defs.      
+
+```Gherkin 
+Scenario: Free subscribers see only the free articles
+Given articles are: 
+| Title           | For Subscription  |
+| Free Article 1  | Free              |
+| Paid Article 1  | Paid              |
+And user is logged in 
+| User Name               | Password          | Subscription  |
+| freeFrieda@example.com  | validPassword123  | Free          |
+When articles are displayed 
+Then articles displayed are 
+| Title           |
+| Free Article 1  | 
+
+Scenario: Subscriber with a paid subscription can access both free and paid articles
+Given articles are: 
+| Title           | For Subscription  |
+| Free Article 1  | Free              |
+| Paid Article 1  | Paid              |
+And user is logged in 
+| User Name               | Password          | Subscription  |
+| paidPattya@example.com  | validPassword123  | Paid          |
+When articles are displayed 
```

Comment:
This step would then become redundant in both scenarios

Reply:
I was split on whether the login should become a separate scenario or not.   I was matching the data in the imperative style

There would at least need to be a step that said the logged user had a paid.free subscription.  Keeping that as a data table emphasizes that Subscription is a domain term.    

I could write a section on splitting scenarios, which might come after this one.